oasislmf/pytools/fm/back_allocation.py

```diff
@@ -0,0 +1,301 @@
+
+            for i in range(child_val_len):
+                if temp_node_extras[p, child_sidx[i], DEDUCTIBLE] < 0:  # realloc loss
+                    if temp_node_extras[p, child_sidx[i], UNDERLIMIT] == 0:
+                        realloc = temp_node_extras[p, child_sidx[i], DEDUCTIBLE] * child_extra[i, DEDUCTIBLE]
+                    else:
+                        realloc = temp_node_extras[p, child_sidx[i], DEDUCTIBLE] * child_extra[i, UNDERLIMIT]
+                    if temp_node_extras[p, child_sidx[i], OVERLIMIT] >= 0:
+                        child_extra[i, OVERLIMIT] = child_extra[i, OVERLIMIT] + temp_node_extras[
+                            p, child_sidx[i], OVERLIMIT] * (child_loss[i] - realloc)
+                    else:
+                        child_extra[i, OVERLIMIT] = - temp_node_extras[p, child_sidx[i], OVERLIMIT] * child_extra[
+                            i, OVERLIMIT]
+
+                    child_loss[i] = (child_loss[i] - realloc) * temp_node_loss[p, child_sidx[i]]
+                    child_extra[i, DEDUCTIBLE] = child_extra[i, DEDUCTIBLE] + realloc
+                    child_extra[i, UNDERLIMIT] = - temp_node_extras[p, child_sidx[i], UNDERLIMIT] * child_extra[
+                        i, UNDERLIMIT]
+
+                else:
+                    if temp_node_extras[p, child_sidx[i], OVERLIMIT] >= 0:
+                        child_extra[i, OVERLIMIT] = child_extra[i, OVERLIMIT] + temp_node_extras[
+                            p, child_sidx[i], OVERLIMIT] * child_loss[i]
+                    else:
+                        child_extra[i, OVERLIMIT] = - temp_node_extras[p, child_sidx[i], OVERLIMIT] * child_extra[
+                            i, OVERLIMIT]
+
+                    if temp_node_extras[p, child_sidx[i], UNDERLIMIT] >= 0:
+                        child_extra[i, UNDERLIMIT] = child_extra[i, UNDERLIMIT] + temp_node_extras[
+                            p, child_sidx[i], UNDERLIMIT] * child_loss[i]
+                    else:
+                        child_extra[i, UNDERLIMIT] = - temp_node_extras[p, child_sidx[i], UNDERLIMIT] * child_extra[
+                            i, UNDERLIMIT]
+
+                    child_extra[i, DEDUCTIBLE] = child_extra[i, DEDUCTIBLE] + temp_node_extras[
+                        p, child_sidx[i], DEDUCTIBLE] * child_loss[i]
+                    child_loss[i] = child_loss[i] * temp_node_loss[p, child_sidx[i]]
+            # print('ba', child['level_id'], child['agg_id'], p, loss_indptr[child['loss'] + p], child_loss[0], temp_node_loss[p, -3],
+            # extras_indptr[child['extra'] + p], child_extra[0])
+
+
+@njit(cache=True, fastmath=True)
+def back_alloc_a2(base_children_len, temp_children_queue, nodes_array, p,
+                  node_val_len, node_sidx, sidx_indptr, sidx_indexes, sidx_val,
+                  loss_in, loss_out, temp_node_loss, loss_indptr, loss_val):
+    """
+    back allocation of loss to the base children
```

Comment:
The function doesn't return anything, so I presume it modifies some arrays in place.
I think it'd be good to make it clear in the docstring, like 
```suggestion
    back allocation of loss to the base children. The function modifies in-place array xyz.
```